ehr/src/org/labkey/ehr/EHRController.java

```diff
@@ -141,9 +143,46 @@ public ApiResponse execute(GetDataEntryItemsForm form, BindException errors)
 
             Collection<DataEntryForm> forms = DataEntryManager.get().getForms(getContainer(), getUser());
             List<JSONObject> formJson = new ArrayList<>();
-            for (DataEntryForm def : forms)
+
+            // restrict forms if the user is in the restricted group
+            var userGroups = SecurityManager.getGroups(getContainer(), getUser());
```

Comment:
I was hoping that this code wouldn't know about the group names. Instead, it would just be asking a filter if a given form is acceptable for the current user.